why-semver.qmd

```diff
@@ -0,0 +1,169 @@
+    it. For instance, there are tools that can automatically generate
+    changelogs based on the commit messages in a repository, that can
+    automatically increment the version number based on the changes
+    made, or that
+:::
+
+::: column
+#### Drawbacks
+
+-   It can be very difficult to determine when something is a major,
+    minor, or patch change. Sometimes it is perfectly clear, but often,
+    a change is hard to place in these three categories.
+-   It can be difficult to determine what constitutes a breaking change.
+-   More difficult (but not impossible) to automate, since it can
+    involve a large amount of human judgement.
+:::
+:::
+
+### CalVer
+
+CalVer is a system that is based on dates. Unlike SemVer, there are more
+variations available to use, for instance, `YYYY.MM` or `YY.MM`. Also
+unlike SemVer, there are more often variations that combine features of
+SemVer and CalVer, such as `YYYY.MINOR.PATCH`.
+
+::: columns
+::: column
+#### Benefits
+
+-   Very easy to use and understand, as it is based on dates. Whatever
+    date the software is released on, is the version number.
+-   Works very well for some software projects, especially those with
+    very regular releases. For instance, the most popular Linux
+    distribution, Ubuntu, uses a form of CalVer, that increment every 6
+    months.
+:::
+
+::: column
+#### Drawbacks
+
+-   It doesn't contain any information on what was changed between
+    versions.
+-   Breaking changes are not communicated in the version number.
+-   There are substantially fewer tools and packages to help automate
+    release and development workflows compared to SemVer.
+-   Doesn't work as well for some software projects, especially those
+    that are not released on a regular basis.
+:::
+:::
+
+## Decision outcome
+
+::: content-hidden
+What decision was made, use the form "We decided on CHOICE because of
+REASONS."
+:::
+
+We decided to use Semantic Versioning (SemVar) for versioning our
+software. While it is more difficult to implement and decide on which
+version to increment based on the changes made, because it is so widely
+used, the availability of tools that support it or exclusively depend on
+it is very large. It will ultimately make it easier for us during
+development because we can make use of those existing tools to handle
+tasks like version release and changelog generation for us.
+
+### Consequences
+
+::: content-hidden
+List some potential consequences of this decision.
+:::
+
+-   One of the biggest consequences of using SemVer is that it does take
+    a bit of effort to implement the tools that will eventually automate
+    many of our tasks for us.
+-   There will also be some variation in how individual team members and
+    contributors label changes, since there is some level of human
+    judgement on categorising changes into the correct version number.
+    There will be some learning within the team based on this.
```

Comment:
```suggestion
One of the biggest consequences of using SemVer is that it does take
a bit of effort to implement the tools that will eventually automate
many of our tasks for us. 

There will also be some variation in how individual team members and
contributors label changes, since there is some level of human
judgement on categorising changes into the correct version number.
This will involve some learning for the team.
```
Removed bullet points and rewrote the last sentence :)